auth.php

```diff
@@ -145,6 +145,26 @@ public function user_signup($user, $notify=true) {
             }
         }
 
+        // Lookup group enrol keys. Not forgetting that group enrolment key is kept in {group}.enrolmentkey.
+        $enrolplugins = $DB->get_records_sql("
+                SELECT e.*, g.enrolmentkey
+                  FROM {groups} g
+                  JOIN {enrol} e ON e.courseid = g.courseid
+                                AND e.enrol = 'self'
+                                AND e.customint1 = 1
+                 WHERE g.enrolmentkey = ?
+        ", array($user->signup_token));
+        foreach ($enrolplugins as $enrolplugin) {
+            if ($enrol->can_self_enrol($enrolplugin) === true) {
+
+                $data = new stdClass();
+                // $data should keep the group enrolment key according to implementation of method $enrol_self_plugin->enrol_self
+                $data->enrolpassword = $enrolplugin->enrolmentkey;
```

Comment:
I though about it too. But then I realised that there Skoda makes some unobvious checking on whether the course enrolment key is different from the group enrolment key in in $enrol_self_plugin->enrol_self at enrol/self/lib.php:160.

So it looks like we have to keep both course and group enrolment keys.

Ultimately Skoda's implementation requires that course enrolment key must be different from group enrolment keys. This should probably be described in Readme.md.